fhir2dataset/fhirpath.py

```diff
@@ -0,0 +1,112 @@
+    }
+})(%(globals)s);
+"""
+
+
+@timing
+def execute(code: str, args: list = None, g: dict = None):
+    """Function to execute code written in javascript
+
+    Args:
+        code (str): javascript code
+        args (list, optional): the arguments that the js code takes as input. Defaults to None.
+        g (dict, optional): the 'this' object in the code. Defaults to None.
+
+    Returns:
+        dict or list: result of the js script
+    """
+    if args is None:
+        args = []
+
+    if g is None:
+        g = {}
+
+    assert isinstance(code, str)
+    assert isinstance(g, dict)
+    assert code.strip(" ").strip("\t").startswith("function"), "Code must be function"
+    prc = Popen(
+        "node",
+        shell=False,
+        stderr=PIPE,
+        stdout=PIPE,
+        stdin=PIPE,
+        cwd="fhir2dataset/metadata",
+        encoding="utf-8",
+    )
+
+    c = wrapper % {"func": code, "globals": json.dumps(g), "args": json.dumps(args)}
+    outs, errs = prc.communicate(input=c)
+    if isinstance(outs, str):
+        outs = json.loads(outs)
+    if "result" in outs:
+        outs = outs["result"]
+        return outs
+    elif "error" in outs:
+        raise Exception((outs.get("error"), errs,))
+    else:
+        raise Exception(errs)
+
+
+@timing
+def multiple_search_dict(resources, fhirpaths):
+    """constructs a list composed of the elements resulting from the fhirpaths contained in the 'fhirpaths' argument applied to an instance of a resource and then concatenates all these lists into a larger list for all resources contained in the 'resources' argument.
+
+    Args:
+        resources (list): json, corresponding to a fhir resource, list 
+        fhirpaths (list): list of fhirpaths to be applied to each resource
+
+    Returns:
+        list: the element list[1][2] corresponds to the element found by the fhirpath at position 1 in the 'fhirpaths' list on the instance at position 2 in the 'resources' list
+    """  # noqa
+    result = execute(
+        """function test(resources, fhirpaths) {
+                const fhirpath = require("fhirpath");
+                const fhirpath_r4_model = require("fhirpath/fhir-context/r4");
+
+                return resources.map((resource) => {
+                    return fhirpaths.map((fhirpath_exp) => {
+                        try {
+                            const result = fhirpath.evaluate(
+                                resource,
+                                fhirpath_exp,
+                                null,
+                                fhirpath_r4_model
+                            );
+                            return result;
+                        } catch (e) {
+                            if (e.message.includes("TypeExpression")) {
+                                const result = [
+                                    "the fhirpath could not be evaluated by the library",
+                                ];
+                                return result;
```

Comment:
you don't have to initialize result variable, you can do : 

return fhirpath.evaluate(resource, fhirpath_exp, null, fhirpath_r4_model);

and 

return ["the fhirpath could not be evaluated by the library"];

Reply:
Right! It's changed :smile:

Reply:
@nriss Why don't you think it's elegant? Maybe by explaining to me why I could find a method that would suit you better :)
Basically the function I use works like a terminal on our computer and therefore can only send and read messages exchanged on this kind of terminal. Would there be another method to write to the terminal from which you run a javascript script?
In any case, it's a good idea to add the explanation in the code smile